feat(helm): add TLS/HTTPS support with cert-manager integration

web-engineer opened the pull request:
- Add TLS configuration to ingress templates (client & server)
- Add tls.enabled and tls.secretName to values.yaml
- Include comprehensive cert-manager documentation in INSTALLATION.md
- Add example annotations for Let's Encrypt cluster issuer
- Support both values.yaml and --set flag configuration methods
- Backward compatible (TLS disabled by default)

Enables automatic HTTPS certificate provisioning when cert-manager is installed in the cluster with appropriate ClusterIssuer configured.

**(Please remove this line only before submitting your PR. Ensure that all relevant items are checked before submission.)** 

## Describe your changes

Briefly describe the changes you made and their purpose. 

## Write your issue number after "Fixes "

Fixes #123 

## Please ensure all items are checked off before requesting a review. "Checked off" means you need to add an "x" character between brackets so they turn into checkmarks.

- [ ] (Do not skip this or your PR will be closed) I deployed the application locally.
- [ ] (Do not skip this or your PR will be closed) I have performed a self-review and testing of my code.
- [ ] I have included the issue # in the PR.
- [ ] I have added i18n support to visible strings (instead of `<div>Add</div>`, use): 
```Javascript
const { t } = useTranslation();
<div>{t('add')}</div>
```
- [ ] I have **not** included any files that are not related to my pull request, including package-lock and package-json if dependencies have not changed
- [ ] I didn't use any hardcoded values (otherwise it will not scale, and will make it difficult to maintain consistency across the application).
- [ ] I made sure font sizes, color choices etc are all referenced from the theme. I don't have any hardcoded dimensions.
- [ ] My PR is granular and targeted to one specific feature.
- [ ] I ran `npm run format` in server and client directories, which automatically formats your code.
- [ ] I took a screenshot or a video and attached to this PR if there is a UI change.



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Optional TLS/HTTPS support for client and server ingress, toggleable via chart values and usable with cert-manager.
  * Pre-deploy validation to catch protocol/TLS mismatches and missing TLS configuration.

* **Documentation**
  * Installation guide expanded with TLS prerequisites, configuration examples, Helm enablement option, and verification steps.

* **Bug Fixes**
  * Early checks to prevent misconfigured ingress/TLS combinations and missing cert-manager issuer annotations.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

web-engineer committed:
feat(helm): add TLS/HTTPS support with cert-manager integration

- Add TLS configuration to ingress templates (client & server)
- Add tls.enabled and tls.secretName to values.yaml
- Include comprehensive cert-manager documentation in INSTALLATION.md
- Add example annotations for Let's Encrypt cluster issuer
- Support both values.yaml and --set flag configuration methods
- Backward compatible (TLS disabled by default)

Enables automatic HTTPS certificate provisioning when cert-manager
is installed in the cluster with appropriate ClusterIssuer configured.

web-engineer commented:
This PR adds optional TLS sections to both ingress templates and introduces client.ingress.tls and server.ingress.tls fields in [values.yaml](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html). It’s backward compatible (off by default). Installation docs include cert-manager/Let’s Encrypt steps and validation commands. Happy to adjust defaults or add Helm schema if desired.

web-engineer committed:
chore(helm): comment out secretName in TLS config, add usage note for cert-manager

web-engineer commented:
Updated: Commented out the `secretName` fields in the TLS config in values.yaml, with a usage note for cert-manager. This keeps the chart minimal by default and clarifies how to enable TLS.

karenvicent commented on charts/helm/checkmate/values.yaml:
```diff
@@ -7,6 +7,13 @@ client:
     host: change_me
     className: nginx
     annotations: {}
+    # Example annotations for cert-manager:
```
While the TLS block has been added, the critical issue regarding the Protocol Mismatch flagged by the AI is still unresolved.

The values.yaml is missing the protocol variable (as shown in the AI suggestion). Without explicitly defining protocol: http (and changing it to https when TLS is enabled), the application will suffer from Mixed Content errors because it will construct internal URLs using HTTP while the Ingress uses HTTPS.

Please add the protocol variable to both Client and Server configurations as suggested.

karenvicent commented on charts/helm/checkmate/templates/client-ingress.yaml:
```diff
@@ -11,6 +11,12 @@ metadata:
   {{- end }}
 spec:
   ingressClassName: {{ .Values.client.ingress.className }}
+  {{- if .Values.client.ingress.tls.enabled }}
+  tls:
+    - hosts:
+        - {{ .Values.client.ingress.host }}
+      secretName: {{ .Values.client.ingress.tls.secretName }}
+  {{- end }}
```
I recommend adding the validation logic suggested by the Llama here.
If tls.enabled is true but the cert-manager.io/cluster-issuer annotation is missing, the chart should fail immediately with a helpful error message. This prevents "silent failures" where the certificate is simply never issued.

web-engineer committed:
feat(helm): enforce TLS + protocol validation and improve docs\n\n- Add prechecks.yaml to validate TLS issuer annotations and secretName\n- Require https when TLS enabled; prevent client https + server http mixed content\n- Validate client.protocol in addition to server.protocol\n- Update INSTALLATION.md to set protocol: https when enabling TLS\n- Clarify values.yaml comments for TLS usage

web-engineer commented:
Addressed feedback: added pre-install validations to prevent silent TLS misconfig (issuer and secretName required, enforce https when TLS is enabled), validated client/server protocol values, and added a guard that fails if client is https but API remains http to prevent mixed content. 

Docs and values comments updated to instruct setting protocol: https when enabling TLS.

web-engineer committed:
chore(helm): default TLS secretName to <release>-* and remove duplicate prechecks file; update docs with protocol flags

web-engineer commented:
mplemented defaults for TLS secret names using release-scoped values, removed duplicate prechecks file, added protocol https flags to --set example, and maintained validations to avoid silent TLS misconfig and mixed-content issues.
